languages/C.groovy

```diff
@@ -0,0 +1,379 @@
+    //compile.dd(new DDStatement().name("TASKLIB").dsn(props."SIGYCOMP_$compilerVer").options("shr"))
+
+    if (buildUtils.isCICS(logicalFile))
+        compile.dd(new DDStatement().dsn(props.SDFHLOAD).options("shr"))
+    if (buildUtils.isSQL(logicalFile)) {
+        compile.dd(new DDStatement().dsn(props.SDSNLOAD).options("shr"))
+        if (props.SDSNEXIT)
+            compile.dd(new DDStatement().dsn(props.SDSNEXIT).options("shr"))
+    }
+
+    if (props.SFELLOAD)
+        compile.dd(new DDStatement().dsn(props.SFELLOAD).options("shr"))
+
+    // adding alternate library definitions
+    if (props.cc_dependenciesAlternativeLibraryNameMapping) {
+        alternateLibraryNameAllocations = buildUtils.parseJSONStringToMap(props.cc_dependenciesAlternativeLibraryNameMapping)
+        alternateLibraryNameAllocations.each { libraryName, datasetDefinition ->
+            datasetName = props.getProperty(datasetDefinition)
+            if (datasetName) {
+                compile.dd(new DDStatement().name(libraryName).dsn(datasetName).options("shr"))
+            }
+            else {
+                String errorMsg = "*! C.groovy. The dataset definition $datasetDefinition could not be resolved from the DBB Build properties."
+                println(errorMsg)
+                props.error = "true"
+                buildUtils.updateBuildResult(errorMsg:errorMsg)
+            }
+        }
+    }
+
+    // add IDz User Build Error Feedback DDs
+    if (props.errPrefix) {
+        compile.dd(new DDStatement().name("SYSADATA").options("DUMMY"))
+        // SYSXMLSD.XML suffix is mandatory for IDZ/ZOD to populate remote error list
+        compile.dd(new DDStatement().name("SYSXMLSD").dsn("${props.hlq}.${props.errPrefix}.SYSXMLSD.XML").options(props.cc_compileErrorFeedbackXmlOptions))
+    }
+
+    // add a copy command to the compile command to copy the SYSPRINT from the temporary dataset to an HFS log file
+    compile.copy(new CopyToHFS().ddName("SYSOUT").file(logFile).hfsEncoding(props.logEncoding))
+
+    return compile
+}
+
+
+/*
+ * createLinkEditCommand - creates a MVSExec xommand for link editing the c object module produced by the compile
+ */
+def createLinkEditCommand(String buildFile, LogicalFile logicalFile, String member, File logFile) {
+
+    String parms = props.getFileProperty('cc_linkEditParms', buildFile)
+    String linker = props.getFileProperty('cc_linkEditor', buildFile)
+    String linkEditStream = props.getFileProperty('cc_linkEditStream', buildFile)
+    String linkDebugExit = props.getFileProperty('cc_linkDebugExit', buildFile)
+
+    // obtain githash for buildfile
+    String cc_storeSSI = props.getFileProperty('cc_storeSSI', buildFile)
+    if (cc_storeSSI && cc_storeSSI.toBoolean() && (props.mergeBuild || props.impactBuild || props.fullBuild)) {
+        String ssi = buildUtils.getShortGitHash(buildFile)
+        if (ssi != null) parms = parms + ",SSI=$ssi"
+    }
+
```

Comment:
The link phase process has been updated with previous zAppBuild releases. 

It is preferred to also make this consistent and add:
* Ability to add IDENTIFY statement - see #369 
* Reference the OBJ dataset, instead of temp dataset allocation - see previous comment
* Leverage additional runtime flags, such as IMS